cmake/BuildFlags.cmake

```diff
@@ -128,7 +128,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC"
   else()
     set(AOT_TARGETS "pvc,bmg,dg2,arl-h,mtl-h,lnl-m")
   endif()
-  if(DEFINED ENV{TORCH_XPU_ARCH_LIST})
```

Comment:
maybe keep this line and add an if clause to the `-device` line? If AOT_TARGETS is an empty string, then skip setting `-device`.